packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedNodeTraversalTest.kt

```diff
@@ -0,0 +1,1390 @@
+@PrepareForTest(Arguments::class)
+@RunWith(RobolectricTestRunner::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+class NativeAnimatedNodeTraversalTest {
+  @get:Rule var rule = PowerMockRule()
+
+  companion object {
+    private val FRAME_LEN_NANOS: Long = 1000000000L / 60L
+    private val INITIAL_FRAME_TIME_NANOS: Long = 14599233201256L /* random */
+  }
+
+  private var frameTimeNanos: Long = 0L
+  private lateinit var reactApplicationContextMock: ReactApplicationContext
+  private lateinit var catalystInstanceMock: CatalystInstance
+  private lateinit var uiManagerMock: UIManagerModule
+  private lateinit var eventDispatcherMock: EventDispatcher
+  private lateinit var nativeAnimatedNodesManager: NativeAnimatedNodesManager
+
+  private fun nextFrameTime(): Long {
+    frameTimeNanos += FRAME_LEN_NANOS
+    return frameTimeNanos
+  }
+
+  @Before
+  fun setUp() {
+    mockStatic(Arguments::class.java)
+    whenever(Arguments.createArray()).thenAnswer { JavaOnlyArray() }
+    whenever(Arguments.createMap()).thenAnswer { JavaOnlyMap() }
+
+    frameTimeNanos = INITIAL_FRAME_TIME_NANOS
+
+    reactApplicationContextMock = mock(ReactApplicationContext::class.java)
+    whenever(reactApplicationContextMock.hasActiveReactInstance()).thenAnswer { true }
+    whenever(reactApplicationContextMock.hasCatalystInstance()).thenAnswer { true }
+    whenever(reactApplicationContextMock.getCatalystInstance()).thenAnswer {
+      catalystInstanceMock
+    }
+    whenever(reactApplicationContextMock.getNativeModule(UIManagerModule::class.java))
+      .thenAnswer { uiManagerMock }
+
+    catalystInstanceMock = mock(CatalystInstance::class.java)
+    whenever(catalystInstanceMock.getJSIModule(any(JSIModuleType::class.java))).thenAnswer {
+      uiManagerMock
+    }
+    whenever(catalystInstanceMock.getNativeModule(UIManagerModule::class.java)).thenAnswer {
+      uiManagerMock
+    }
+
+    uiManagerMock = mock(UIManagerModule::class.java)
+    eventDispatcherMock = mock(EventDispatcher::class.java)
+    whenever(uiManagerMock.getEventDispatcher()).thenAnswer { eventDispatcherMock }
+    whenever(uiManagerMock.getConstants()).thenAnswer {
+      MapBuilder.of("customDirectEventTypes", MapBuilder.newHashMap<Any, Any>())
+    }
+    whenever(uiManagerMock.getDirectEventNamesResolver()).thenAnswer {
+      object : UIManagerModule.CustomEventNamesResolver {
+        override fun resolveCustomEventName(eventName: String): String {
+          val directEventTypes: Map<String, Map<String, String>>? =
+            uiManagerMock?.constants?.get("customDirectEventTypes")
+              as? Map<String, Map<String, String>>?
+          if (directEventTypes != null) {
+            val customEventType: Map<String, String>? =
+              directEventTypes[eventName] as? Map<String, String>?
+            if (customEventType != null) {
```

Comment:
This can be simplified (i.e. no need to `null` check those fields are you're creating them). cc @rshest if you want to integrate this change